SingularityService/src/main/java/com/hubspot/singularity/data/RequestManager.java

```diff
@@ -208,7 +213,7 @@ public SingularityCreateResult update(SingularityRequest request, long timestamp
 
   public SingularityCreateResult save(SingularityRequest request, RequestState state, RequestHistoryType eventType, long timestamp, Optional<String> user, Optional<String> message) {
     saveHistory(new SingularityRequestHistory(timestamp, user, eventType, request, message));
-
+    leaderCache.putRequest(new SingularityRequestWithState(request, state, timestamp));
```

Comment:
The `DELETE` for a request right now is actually just a history update, we never actually remove the node. All the getRequests calls are fetching 'deleted' things still, they are just generally filtered out based on their current state